tests/chainer_tests/serializers_tests/test_npz.py

```diff
@@ -445,6 +445,30 @@ def test_load_without_path(self):
             target.parent_linear.W.data)
 
 
+class TestPickle(unittest.TestCase):
+    def test_pickle(self):
+        obj = ['data', object()]
+
+        file = six.BytesIO()
+        l = link.Chain()
+        l.add_persistent('obj', obj)
+        npz.save_npz(file, l)
+
+        # Should success if `allow_pickle=True`.
+        file.seek(0)
+        l.obj[0] = 'overwritten'
+        npz.load_npz(file, l, allow_pickle=True)
+        assert l.obj[0] == 'data'
+
+        # Should raise an error if `allow_pickle=False` (default).
+        file.seek(0)
+        self.assertRaises(
```

Comment:
How about `pytest.raises`?

Reply:
I fixed to use `with self.assertRaises()`.
Are there merits of `pytest.raises` over `self.assertRaises`?

Reply:
It's about consistency, as it's pytest's standard way and we're moving towards pytest.

Reply:
I thought mixing two kinds of notation in the same codebase is rather inconsistent and confusing for users/contributors, considering this is an open source project.

I'm also worrying that by strongly depending on one test framework makes future migration a tough work (considering migration from nose to pytest was pretty easy because our test code does not strongly depend on nose).

Reply:
I understand the points, and actually I'm not confident about that.
Maybe we should make a clear policy.

There are some conterarguments about those points:
* It's inevitable to mix different conventions if we're moving one convention to another.
* We can say using `assert` also depends on pytest.

Reply:
As for `pytest.raises` specifically, there's one advantage over `self.assertRaises`: no need of `self`. If the checking logic is factored out from the `TestCase` class (there may be many reasons to do so), we need to pass around `self` instance, if we stick with the old convention.

Reply:
I come up with the simple policy:

* Keep avoiding `import pytest` in our test code base (as we did in nose)
* Permit use of simple `assert`

Thoughts behind the policy above is that I want to minimize the requirements of test framework when we have to choose another framework to migrate.

* Why avoid `import pytest`?
    * Once we allow `import pytest`, we allow every feature in pytest to be used in future test code.
    * This rule is simple enough for contributors.
* Why simple `assert` can be a exception:
    * It is shorter than `self.assertEqual` and also improves readability. (= cost-effective)
    * Even with simple assert, we can say that the test code is still compatible with standard `unittest` (although without nice dump of arguments)

c.f., In NumPy, it seems that they are paying much effort to abstracting out test framework by their own and gradually moving from nose to pytest.

* https://github.com/numpy/numpy/issues/9352#issue-240027620
* https://github.com/numpy/numpy/tree/master/numpy/testing has pytest_tools and nose_tools, and `assert_raises` is defined for each framework.
  * https://github.com/numpy/numpy/blob/master/numpy/testing/pytest_tools/utils.py#L1190
  * https://github.com/numpy/numpy/blob/master/numpy/testing/nose_tools/utils.py#L1208

Reply:
I made an issue for this topic: #4557.
Let's keep `self.assertRaises` in this PR.

Reply:
Thanks!